docs/api/web-frame-main.md

```diff
@@ -142,6 +142,29 @@ ipcRenderer.on('port', (e, msg) => {
 })
 ```
 
+#### `frame.collectJavaScriptCallStack()` _Experimental_
+
+Returns `Promise<string> | Promise<void>` - A promise that resolves with the currently running JavaScript call
+stack. If no JavaScript runs, the promise will never resolve. If the call stack is unable to be collected, it
```

Comment:
> If no JavaScript runs, the promise will never resolve

Just curious, what is the better way to determine this case ? would checking for [current stack position](https://source.chromium.org/chromium/chromium/src/+/main:v8/src/utils/utils.cc;l=222) against the threads stack base address help ?

Reply:
> would checking for [current stack position](https://source.chromium.org/chromium/chromium/src/+/main:v8/src/utils/utils.cc;l=222) against the threads stack base address help ?

I think this would be best left as feedback (or a CL) upstream so it can be fixed there. We should mention this in https://issues.chromium.org/issues/40268201

> Naively, would providing a timeout argument be helpful to provide users with an escape hatch?

If a fix cannot be applied upstream, I think we should consider this. My thinking is we merge this as experimental for now and revisit after seeing results from usage.